daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-features-concepts.md

```diff
@@ -28,9 +28,179 @@ There are several different kinds of tasks that a workflow can schedule, includi
+		if err := ctx.GetInput(&name); err != nil {
+			return nil, err
+		}
+		return fmt.Sprintf("Hello, %s!", name), nil
+	})
+
+	ctx := context.Background()
+	client, engine := startEngine(ctx, r)
+
+	instanceID := api.InstanceID("IGNORE_IF_RUNNING_OR_COMPLETED")
+	reuseIDPolicy := &api.OrchestrationIdReusePolicy{
+		Action:          api.REUSE_ID_ACTION_IGNORE,
+		OperationStatus: []api.OrchestrationStatus{api.RUNTIME_STATUS_RUNNING, api.RUNTIME_STATUS_COMPLETED, api.RUNTIME_STATUS_PENDING},
+	}
+
+	// Run the orchestration.
+	id, err := client.ScheduleNewOrchestration(ctx, "SingleActivity", api.WithInput("World"), api.WithInstanceID(instanceID))
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	// Wait for orchestration to start...
+	client.WaitForOrchestrationStart(ctx, id)
+	// Schedule the workflow again using the same id. However it will error out since it already exists. 
+	id, err = client.ScheduleNewOrchestration(ctx, "SingleActivity", api.WithInput("World"), api.WithInstanceID(id), api.WithOrchestrationIdReusePolicy(reuseIDPolicy))
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+}
+```
+
+
+**Example 2**
+
+The following example demonstrates `main` executing a workflow twice. In the example:
+1. The workflow calls a single activity with orchestration ID reuse policy.
+1. The reuse ID policy specifies the action `IGNORE_IF_RUNNING_OR_COMPLETED` and the target statuses of `RUNNING`, `COMPLETED`, `PENDING`. 
+1. The second call to create a workflow with the same instance ID is expected to be ignored if the first workflow instance is one of the target statuses.
+
+```go
+func main() {
+	r := task.NewTaskRegistry()
+	r.AddOrchestratorN("SingleActivity", func(ctx *task.OrchestrationContext) (any, error) {
+		var input string
+		if err := ctx.GetInput(&input); err != nil {
+			return nil, err
+		}
+		var output string
+		err := ctx.CallActivity("SayHello", task.WithActivityInput(input)).Await(&output)
+		return output, err
+	})
+	r.AddActivityN("SayHello", func(ctx task.ActivityContext) (any, error) {
+		var name string
+		if err := ctx.GetInput(&name); err != nil {
+			return nil, err
+		}
+		return fmt.Sprintf("Hello, %s!", name), nil
+	})
+
+	ctx := context.Background()
+	client, engine := startEngine(ctx, r)
+
+	instanceID := api.InstanceID("IGNORE_IF_RUNNING_OR_COMPLETED")
+	reuseIDPolicy := &api.OrchestrationIdReusePolicy{
+		Action:          api.REUSE_ID_ACTION_IGNORE,
+		OperationStatus: []api.OrchestrationStatus{api.RUNTIME_STATUS_RUNNING, api.RUNTIME_STATUS_COMPLETED, api.RUNTIME_STATUS_PENDING},
+	}
+
+	// Run the orchestration.
+	id, err := client.ScheduleNewOrchestration(ctx, "SingleActivity", api.WithInput("World"), api.WithInstanceID(instanceID))
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	// Wait for orchestration to start...
+	client.WaitForOrchestrationStart(ctx, id)
+	// Schedule the workflow again using the same id. However it will ignore creating the new orchestration, since the id is already in use.
```

Comment:
What do you mean "ignore". You mean nothing happens (which is strange), or it will eventually get rescheduled. A silent failure event would be really hard to determine. We need to be clearer here on what happens IMO. Can you explain this more and why I would choose this option?

Reply:
@kaibocai - See my questions. It would help to clarify clearly when these actions are used.
Action: api.REUSE_ID_ACTION_IGNORE

Reply:
> What do you mean "ignore". You mean nothing happens (which is strange), or it will eventually get rescheduled. A silent failure event would be really hard to determine. We need to be clearer here on what happens IMO. Can you explain this more and why I would choose this option?

@kaibocai an explanation of this would help me as well!

Reply:
so when you want to start a orchestration with the ID that not being used by any running orchestration then you can use this option. More discussion details can be found https://github.com/microsoft/durabletask-go/issues/42

Reply:
Ok, but this is a discussion and there is no final conclusion here of what was actually implemented. Just like Oli created here, we need a table for the reuse ID options and what each one does. Can you add this?